portfolio/src/main/java/com/google/sps/servlets/CommentsStatisticServlet.java

```diff
@@ -0,0 +1,73 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.FetchOptions.Builder;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.QueryResultList;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.lang.Integer;
+import java.sql.Timestamp;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles comments statistic request. */
+@WebServlet("/comments-stats")
+public class CommentsStatisticServlet extends HttpServlet {
+  
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private DateTimeFormatter dateFormatter = DateTimeFormatter.ISO_LOCAL_DATE;
+  private ZoneId zoneId = ZoneId.systemDefault();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String[] dateStrings = (String[]) request.getParameterValues("date");
+    Map<String, Integer> commentsCount = new HashMap<>();
+    for (String dateString: dateStrings) {
+      LocalDate date;
+      try {
+        date = LocalDate.parse(dateString, dateFormatter);
+      } catch(DateTimeParseException e) {
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      int count = getNumberOfCommentsOnDate(date);
```

Comment:
Could you do this with just one database call instead of a separate call for each day? Putting several similar queries together gives the datastore a chance to make optimizations. For example, instead of querying for a single day, query for the entire range of 7 days and then count each day in the query response.